core/unit_test/tools/include/ToolTestingUtilities.hpp

```diff
@@ -153,7 +131,7 @@ struct function_traits<R (*)(A...)> {
   constexpr static int num_arguments = sizeof...(A);
   template <class Call, class... Args>
   static auto invoke_as(const Call& call, Args&&... args) {
-    if (!are_valid(std::dynamic_pointer_cast<A>(std::forward<Args>(args))...)) {
+    if ((!std::dynamic_pointer_cast<A>(std::forward<Args>(args)) && ...)) {
```

Comment:
This change looks like a mixup of where the negation is applied. It's now testing that all the pointers are the wrong type, rather than any of them

Reply:
Thx.  It's interesting that virtually all the builds had all test passings with that bug...